src/averageAges.js

```diff
@@ -37,7 +43,32 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people
+    .filter(person => person.sex === 'f');
+
+  const mother = [];
+
+  for (const woman of women) {
+    if (people.find(person => person.mother === woman.name)) {
+      mother.push(woman);
+    }
+  }
+
+  const motherNumber = mother.length;
+
+  let avgAge;
+
+  if (withChildren) {
+    const ages = mother.map(mom => mom.died - mom.born);
+
+    avgAge = ages.reduce((a, b) => a + b, 0) / motherNumber;
+  } else {
+    const ages = women.map(mom => mom.died - mom.born);
+
+    avgAge = ages.reduce((a, b) => a + b, 0) / women.length;
+  }
```

Comment:
1. filter people by the condition (withChildren or not)
2. find average ages of filtered array